### PR TITLE
[material] Add oneOf tests

### PR DESCRIPTION
**Pull request by edgarmueller**
Adds initial set of tests for oneOf renderer, fix default value initialization

### CODE REVIEW

File: packages/material/test/renderers/MaterialOneOfRenderer.test.tsx [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+      wrapper.update();
+
+      // click add button
+      buttons = wrapper.find('button');
+      const addButton = buttons.at(2);
+      const nrOfRowsBeforeAdd = wrapper.find('tr');
+      addButton.simulate('click');
+      addButton.simulate('click');
+
+      // check if row has been added
```
Comment (eneufeld) [inline packages/material/test/renderers/MaterialOneOfRenderer.test.tsx]: this should say 'check if row**s** have been added as you added 2

File: packages/material/test/renderers/MaterialOneOfRenderer.test.tsx [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+      const nrOfRowsAfterAdd = wrapper.find('tr');
+
+      // 1 header row + no data row
+      expect(nrOfRowsBeforeAdd.length).toBe(2);
+      // 1 header row + 2 data rows (one is replacing the 'No data' one)
+      expect(nrOfRowsAfterAdd.length).toBe(3);
+    });
+  });
+
+  it('should add an object within an array', () => {
```
Comment (eneufeld) [inline packages/material/test/renderers/MaterialOneOfRenderer.test.tsx]: isn't this exactly the same as the previous except one more assert?

File: packages/material/test/renderers/MaterialOneOfRenderer.test.tsx [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+  return store;
+};
+
+describe('Material oneOf renderer', () => {
+
+  let wrapper: ReactWrapper;
+
+  afterEach(() => wrapper.unmount());
+
+  it('should add an item at correct path', () => {
```
Comment (eneufeld) [inline packages/material/test/renderers/MaterialOneOfRenderer.test.tsx]: can't we unify this with the `should switch to array based oneOf subschema, then switch back, then edit` test`? imho  the second one is a better test of the first one especially if you remove the intermediate objects which is tested in the array tests

File: packages/material/test/renderers/MaterialOneOfRenderer.test.tsx (hunk 2 above)
Comment (edgarmueller) [reply packages/material/test/renderers/MaterialOneOfRenderer.test.tsx]: Yes, it's only the schema that differs.

File: packages/material/test/renderers/MaterialOneOfRenderer.test.tsx (hunk 1 above)
Comment (edgarmueller) [reply packages/material/test/renderers/MaterialOneOfRenderer.test.tsx]: Depends on the view point: initially we already have two rows (header + no data), if we add two items, one is replacing the no data one, hence there's one row more than initially. But I can update if you want, if 'rows' seems to make more sense to you.

File: packages/material/test/renderers/MaterialOneOfRenderer.test.tsx (hunk 3 above)
Comment (edgarmueller) [reply packages/material/test/renderers/MaterialOneOfRenderer.test.tsx]: Yes, we could do that, the schema is different though. Whatever makes more sense to you.

File: packages/material/test/renderers/MaterialOneOfRenderer.test.tsx (hunk 1 above)
Comment (eneufeld) [reply packages/material/test/renderers/MaterialOneOfRenderer.test.tsx]: I'm fine with both, my thinking was that you had no data, and add two entries, but I see what you meant.

File: packages/material/test/renderers/MaterialOneOfRenderer.test.tsx (hunk 3 above)
Comment (eneufeld) [reply packages/material/test/renderers/MaterialOneOfRenderer.test.tsx]: I now see the different schemas, but I just tried out using one schema for both tests and they still work. The array part which is the only one being checked is the same for both schemas.

File: packages/material/test/renderers/MaterialOneOfRenderer.test.tsx (hunk 2 above)
Comment (eneufeld) [reply packages/material/test/renderers/MaterialOneOfRenderer.test.tsx]: I now see the different schemas, but I just tried out using one schema for both tests and they still work. The array part which is the only one being checked is the same for both schemas.

**Comment by edgarmueller** [comment]
@eneufeld I've extracted some utility methods to reduce boilerplate, but kept the tests as is otherwise.

File: packages/material/src/complex/MaterialOneOfRenderer.tsx [hunk 4]
```diff
@@ -69,7 +69,8 @@ class MaterialOneOfRenderer extends React.Component<ControlProps & OneOfProps, M
     });
   };
 
-  handleChange = (_event: any, newOneOfIndex: number) => {
+  // TODO: rename
```
Comment (eneufeld) [inline packages/material/src/complex/MaterialOneOfRenderer.tsx]: the todo is obsolete now

Comment (edgarmueller) [reply packages/material/src/complex/MaterialOneOfRenderer.tsx]: Thanks, fixed

**Pull request merged**